server/services/config/http.go

```diff
@@ -65,11 +65,17 @@ func (h *http) Fetch(ctx context.Context, forge forge.Forge, user *model.User, r
 
 	status, err := h.client.Send(ctx, net_http.MethodPost, h.endpoint, body, response)
 	if err != nil && status != 204 {
-		return nil, fmt.Errorf("failed to fetch config via http (%d) %w", status, err)
+		return nil, fmt.Errorf("failed to fetch config via http from endpoint %s (status: %d): %w", h.endpoint, status, err)
 	}
 
+	// handle 204 - no new config available, return old config with error
+	if status == net_http.StatusNoContent {
+		return oldConfigData, fmt.Errorf("config endpoint returned 204 No Content, using fallback config")
```

Comment:
This shouldn't return an error I think. Just
```suggestion
		return oldConfigData, nil
```

Or did you test it with this and it didn't work?

Because here there is no difference between status 204 and a failing status (line 78).